sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntityEffectImpl.scala

```diff
@@ -63,70 +63,69 @@ class EventSourcedEntityEffectImpl[S] extends Builder[S] with OnSuccessBuilder[S
     secondary
   }
 
-  override def emitEvent(event: Any): OnSuccessBuilder[S] = {
+  override def emitEvent(event: Any): EventSourcedEntityEffectImpl[S] = {
```

Comment:
Hmm, isn't the whole point of working with a builder to have a two step thing where you first have to specify the primary effect, get a OnSuccessBuilder back which limits the possible additional effects to the secondary ones (`emitEvent(event).thenReply(state => some reply)`) or is that solved some way in the public API I'm missing?

Reply:
I'm just not narrowing down the type so I can delegate to it.

Reply:
Yeah, I just realised that the original trait still forces the user to follow the builder. Though, I don't see why we needed to change it. We haven't done so for the others.

Reply:
Just to add @octonato and I talked about it, it's for some easy wrapping.

Reply:
Yep, makes sense and users won't see it. Akka Projections does the same trick in a few places, but for other reasons.